xbmc/cores/AudioEngine/Sinks/AESinkStarfish.cpp

```diff
@@ -0,0 +1,350 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "AESinkStarfish.h"
+
+#include "CompileInfo.h"
+#include "utils/JSONVariantWriter.h"
+#include "utils/log.h"
+#include "xbmc/cores/AudioEngine/AESinkFactory.h"
+
+#include <chrono>
+#include <thread>
+
+#include <nlohmann/json.hpp>
+
+void CAESinkStarfish::Register()
+{
+  AE::AESinkRegEntry entry;
+  entry.sinkName = "Starfish";
+  entry.createFunc = CAESinkStarfish::Create;
+  entry.enumerateFunc = CAESinkStarfish::EnumerateDevicesEx;
+  AE::CAESinkFactory::RegisterSink(entry);
+}
+
+std::unique_ptr<IAESink> CAESinkStarfish::Create(std::string& device, AEAudioFormat& desiredFormat)
+{
+  auto sink = std::make_unique<CAESinkStarfish>();
+  if (sink->Initialize(desiredFormat, device))
+    return sink;
+
+  return {};
+}
+
+void CAESinkStarfish::EnumerateDevicesEx(AEDeviceInfoList& list, bool force)
+{
+  CAEDeviceInfo info;
+  info.m_deviceName = "Starfish";
+  info.m_displayName = "Starfish (Passthrough only)";
+  info.m_channels = AE_CH_LAYOUT_5_1;
+  info.m_wantsIECPassthrough = false;
+
+  // PCM disabled for now as the latency is just too high, needs more research
+  // Thankfully, ALSA or PulseAudio do work as an alternative for PCM content
+  /*info.m_dataFormats.push_back(AE_FMT_U8);
+  info.m_dataFormats.push_back(AE_FMT_S16NE);
+  info.m_dataFormats.push_back(AE_FMT_S16LE);
+  info.m_dataFormats.push_back(AE_FMT_S16BE);
+  info.m_dataFormats.push_back(AE_FMT_S32NE);
+  info.m_dataFormats.push_back(AE_FMT_S32LE);
+  info.m_dataFormats.push_back(AE_FMT_S32BE);
+  info.m_dataFormats.push_back(AE_FMT_FLOAT);
+  info.m_dataFormats.push_back(AE_FMT_DOUBLE);*/
+  info.m_dataFormats.push_back(AE_FMT_RAW);
+
+  info.m_deviceType = AE_DEVTYPE_IEC958;
+  info.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_AC3);
+  //info.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_EAC3);
+  //info.m_streamTypes.push_back(CAEStreamInfo::STREAM_TYPE_TRUEHD);
+
+  info.m_sampleRates.push_back(48000);
+  info.m_sampleRates.push_back(44100);
+  info.m_sampleRates.push_back(32000);
+  info.m_sampleRates.push_back(24000);
+  info.m_sampleRates.push_back(22050);
+  info.m_sampleRates.push_back(16000);
+  info.m_sampleRates.push_back(12000);
+  info.m_sampleRates.push_back(8000);
+
+  list.push_back(info);
+}
+
+CAESinkStarfish::CAESinkStarfish() : m_starfishMediaAPI(std::make_unique<StarfishMediaAPIs>())
+{
+}
+
+CAESinkStarfish::~CAESinkStarfish()
+{
+}
+
+bool CAESinkStarfish::Initialize(AEAudioFormat& format, std::string& device)
```

Comment:
error paths (return false) in this function could use some log messages for easier debugging